packages/amplify-graphql-auth-transformer/src/__tests__/searchable-auth.test.ts

```diff
@@ -110,6 +110,57 @@ test('auth logic is enabled for iam/apiKey auth rules', () => {
   );
 });
 
+test('auth logic is enabled for identityPool/apiKey auth rules', () => {
+  const expectedDirectives = ['aws_api_key', 'aws_iam'];
+  const validSchema = `
+        type Post @model
+            @searchable
+            @auth(rules: [
+                { allow: public, provider: apiKey } # api key is allowed
+                { allow: private, provider: identityPool } # auth roles are allowed
+            ]) {
+            id: ID!
+            content: String
+            secret: String @auth(rules: [{ allow: private, provider: identityPool }]) # only auth role can do crud on this
+        }
+    `;
+  const authConfig: AppSyncAuthConfiguration = {
+    defaultAuthentication: {
+      authenticationType: 'AMAZON_COGNITO_USER_POOLS',
+    },
+    additionalAuthenticationProviders: [
```

Comment:
What if the model contains only API KEY rule, do we still expect IAM policy based auth to work with those models?

Reply:
Only if new IAM behavior is enabled. (It's going to be boolean flag at data construct level).

Reply:
See also https://app.asana.com/0/1206048977824382/1206486113291633/f .

Reply:
In that case, please add a test with API_KEY as default authentication and IAM as additional authentication provider. In such cases, we need to add AppSync aws_iam directive to all models (may be on all fields too - IDR exact implementation).

Reply:
@sundersc The tests in this file seem to already assert `aws_iam` presence. So I just added extra `authConfig` permutations to iam/identityPool tests here https://github.com/aws-amplify/amplify-category-api/pull/2320/commits/b8fe34179b020852701f36872148141148445924 .

Reply:
> Only if new IAM behavior is enabled.
Is that in scope of this PR?

Reply:
> Only if new IAM behavior is enabled.

No this will be separate change. This PR is focused only on aliasing existing functionality.